packages/maven.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+# Install a custom Maven version - https://maven.apache.org
+#
+# To run this script on Codeship, add the following
+# command to your project's setup commands:
+# \curl -sSL https://raw.githubusercontent.com/codeship/scripts/master/packages/maven.sh | bash -s
+#
+# Add the following environment variable to your project configuration
+# (otherwise the default below will be used).
+# * MAVEN_VERSION
+#
+MAVEN_VERSION=${MAVEN_VERSION:="3.5.2"}
+MAVEN_DIR=${MAVEN_DIR:=$HOME/cache/maven-$MAVEN_VERSION}
```

Comment:
Do we know how big that directory usually gets? If it gets too big, the cache will be rejected, which would be bad.

Maybe we can only put a subdirectory of the `MAVEN_DIR` on the cache, for caching the dependencies (in a packaged state), but not the complete tree.